test/integration/custom-routes/test/index.test.js

```diff
@@ -312,6 +312,11 @@ const runTests = (isDev = false) => {
     expect(res.headers.get('somefirst')).toBe('hi')
   })
 
+  it('should support URL for header key/values', async () => {
+    const res = await fetchViaHTTP(appPort, '/without-params/url')
+    expect(res.headers.get('x-url')).toBe('https://example.com')
+  })
```

Comment:
Let's also add a test for this case https://github.com/zeit/next.js/pull/12031/files#r417555141